### PR TITLE
feat: Disable editing team settings in `jx edit buildpack`

### LINKED ISSUE #7126: The command "jx edit buildpack" does nothing
### Summary 
Following [our Slack discussion](https://kubernetes.slack.com/archives/C9MBGQJRH/p1588181950110200?thread_ts=1588089186.068200&cid=C9MBGQJRH), I've tried to configure jenkins-x with custom buildpacks, but it didn't work, no change at all.

```
$ jx edit buildpack -u https://github.com/my-org/jenkins-x-kubernetes
? Defaulting workload build pack: Kubernetes Workloads: Automated CI+CD with GitOps Promotion
Setting the team build pack to kubernetes-workloads repo: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes.git ref: master
```

### Steps to reproduce the behavior
- fork the jenkins-x-kubernetes buildpack repository
- execute this command:
```
$ jx edit buildpack -u https://github.com/my-org/jenkins-x-kubernetes
```

### Expected behavior
JX is configured to use the custom buildpacks, and this configuration is saved in the jx boot dev repository.
### Actual behavior
Nothing happens.
### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.1284
Kubernetes cluster v1.15.7
kubectl            v1.17.3
helm client        2.16.3
git                2.21.1 (Apple Git-122.3)
Operating System   Mac OS X 10.15.3 build 19D76
```
### Kubernetes cluster

AKS cluster, manually created.

@vfarcic FYI
Relate to https://github.com/jenkins-x/jx/issues/6374

### PR DESCRIPTION
**Pull request by mrageh**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
This pull request removes all the code that edits Team settings in `jx edit buildpack`, since that no longer works after the introduction of [`jx boot`](https://github.com/jenkins-x/jx/issues/7126#issuecomment-628537697).

#### Which issue this PR fixes

fixes #7126

### CODE REVIEW

**Comment by mrageh** [comment]
@hferentschik none, but I worry about just removing a command from JX that's been there for so long. Is that something we want to do? I'd personally be happy to remove it :)

File: pkg/cmd/edit/edit_buildpack.go [hunk 1]
```diff
@@ -20,20 +17,11 @@ import (
...
-		jx edit buildpack
-
-        # to switch to classic workloads for your team
-		jx edit buildpack -n classic-workloads
-
-        # to switch to kubernetes workloads for your team
-		jx edit buildpack -n kubernetes-workloads
-		
-		For more documentation see: [https://jenkins-x.io/architecture/build-packs/](https://jenkins-x.io/architecture/build-packs/)
+		For more info see documentation: [https://jenkins-x.io/architecture/build-packs/](https://jenkins-x.io/architecture/build-packs/)
```
[the remark this answers is not in the document]
Comment (mrageh) [reply pkg/cmd/edit/edit_buildpack.go:24]: 👍

**Comment by hferentschik** [comment]
> none, but I worry about just removing a command from JX that's been there for so long.

I cannot see how it would be helpful to anyone. Just confusing. 

> Is that something we want to do?

I would. Let's hear what others think and take it from there.

**Comment by ankitm123** [comment]
Should we not deprecate it, and then remove it?

**Comment by mrageh** [comment]
That's in essence what this PR is trying to do, there's another docs PR that adds it to the deprecation list

**Comment by mrageh** [comment]
However @hferentschik makes a good point, why go through the deprecation cycle? What's the the benefit for end users? The command does not work, so no one can be using it anyway. So why keep it around and deprecate it, when none of our users can depend on it?

**Comment by mrageh** [comment]
@hferentschik I've removed the edit buildpack command, your reasoning for removing it is sound.

**Comment by vfarcic** [comment]
`jx edit buildpack` does work with `jx install` which is still around and I'd say it should stay around until JX Boot works (or is documented) with the `kubernetes` provider as a catch-all solution. For now, `jx install` is an option for all those that are not using GKE and EKS. Even if `jx boot` does work with the `kubernetes` provider, there is no mention of that in the docs (or is not easy to find).

**Comment by hferentschik** [comment]
@mrageh, I am closing this pull request as it stands now. As discussed, we should add a check for the type of cluster we have and:

- if `jx install` based cluster keeps things as they are
- if `jx boot` based cluster print a warning (maybe with a link to some docs) and exit w/o doing anything

**Pull request closed without merging**